client/config/config.go

```diff
@@ -18,60 +20,79 @@ func DefaultConfig() *ClientConfig {
 	}
 }
 
-type ClientConfig struct {
+// ClientConfig is an alias for Config for backward compatibility
+// Deprecated: use Config instead which avoid name stuttering
+type ClientConfig Config
+
+type Config struct {
 	ChainID        string `mapstructure:"chain-id" json:"chain-id"`
 	KeyringBackend string `mapstructure:"keyring-backend" json:"keyring-backend"`
 	Output         string `mapstructure:"output" json:"output"`
 	Node           string `mapstructure:"node" json:"node"`
 	BroadcastMode  string `mapstructure:"broadcast-mode" json:"broadcast-mode"`
 }
 
-func (c *ClientConfig) SetChainID(chainID string) {
-	c.ChainID = chainID
-}
-
-func (c *ClientConfig) SetKeyringBackend(keyringBackend string) {
-	c.KeyringBackend = keyringBackend
-}
-
-func (c *ClientConfig) SetOutput(output string) {
-	c.Output = output
-}
-
-func (c *ClientConfig) SetNode(node string) {
-	c.Node = node
-}
-
-func (c *ClientConfig) SetBroadcastMode(broadcastMode string) {
-	c.BroadcastMode = broadcastMode
+// ReadFromClientConfig reads values from client.toml file and updates them in client.Context
+// It uses CreateClientConfigAndContext internally with no custom template and custom config.
+func ReadFromClientConfig(ctx client.Context) (client.Context, error) {
+	return CreateClientConfigAndOrContext(ctx, "", nil)
 }
 
-// ReadFromClientConfig reads values from client.toml file and updates them in client Context
-func ReadFromClientConfig(ctx client.Context) (client.Context, error) {
+// CreateClientConfigAndContext reads the client.toml file and returns a new populated client.Context
```

Comment:
The name is long but defines better what the function does than how `ReadFromClientConfig` was named.